### ISSUE TITLE
Add data sources to get all BigQuery datasets in a project and all BigQuery tables in a dataset

### ISSUE THREAD

**Issue by yu-iskw** [hashicorp/terraform-provider-google#17695]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team) to expedite investigation and resolution of this issue.

### Description

A dataset resource to get all datasets in a project and to get all tables in a dataset would be useful so that we can dynamically pass datasets to other resources. For instance, the `google_data_loss_prevention_job_trigger` resource enables us to run jobs for DLP in BigQuery. We pass table refere to the resource. If it is possible to get all datasets and tables with data sources, we can dynamically pass the information to the resource.

### New or Affected Resource(s)

* data.google_bigquery_datasets
* data.google_bigquery_tables

## `data.google_bigquery_datasets`

```tf 
data "google_project" "project1" {
}

data "google_bigquery_datasets" "project1" {
  project = data.google_project.project1.project_id
}
```

## `data.google_bigquery_tables`
```
data "google_bigquery_tables" "tables_in_dataset1" {
  project = data.google_project.project1.project_id
  dataset_id = "sample_dataset"
}
```

### References

* https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets/list
* https://cloud.google.com/bigquery/docs/reference/rest/v2/tables/list
* https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/data_loss_prevention_job_trigger

b/331668413

**Comment by anubhavsharma515** [comment]
Hey! Is it possible for me to contribute to this?

**Comment by wj-chen** [comment]
> Hey! Is it possible for me to take this enhancement up?

Much appreciated your contribution! Let's discuss more in your PRs.

**Comment by billinghamj** [comment]
@SarahFrench I don't think this is completed - it looks like that PR added the ability to list tables, but not to list datasets?

**Comment by SarahFrench** [comment]
A data source for tables was added but there isn't a data source for data sets yet. The author of that last PR might [open a new PR to implement the dataset data source](https://github.com/GoogleCloudPlatform/magic-modules/pull/11552#pullrequestreview-2278675204), but if that doesn't happen then that PR's commit history has some work that others could use to open their own PR
